### ISSUE TITLE
Perform stress testing on titiler API

### ISSUE THREAD

**Issue by amarouane-ABDELHAK** [NASA-IMPACT/veda-stac-ingestor#68]

**Comment by amarouane-ABDELHAK** [comment]
The overall goal of this issue is to:

- Design and execute stress tests on the titiler API to simulate high-stress scenarios.

- Measure and analyze the API's performance metrics, including response times, throughput, error rates, and resource utilization.

- Identify any performance bottlenecks or scalability limitations within the titiler API.

- Define a strategy to optimize the API to enhance its scalability, reliability, and response times.

- Document the stress testing process, results, and optimization strategies for future reference.

**Comment by j08lue** [comment]
I have been slow to respond to this, @amarouane-ABDELHAK - some people at Development Seed team have in the past been designing performance tests for dynamic tiling (TiTiler), most recently @abarciauskas-bgse.

I'd be happy to try to make the connection to their work.

> Document the stress testing process, results, and optimization strategies for future reference

How far are you at this point - what types of tests do you have planned / have you already conducted? Is this ticket already work-in-progress, actually?

**Comment by amarouane-ABDELHAK** [comment]
Yes this ticket is `in-progress` I made a plan and conducted some load testing on STAC api and Titiler api, I will document the findings here tomorrow (didn't get the chance to so today)
Until then I will share the reports here as a reference: 
[stac api report](http://veda-fcx.s3-website-us-west-2.amazonaws.com/reports/_api_stress_test_report.html)
[raster api report](http://veda-fcx.s3-website-us-west-2.amazonaws.com/reports/raster_api_stress_test_report.html)
